skbio/alignment/_indexing.py

```diff
@@ -0,0 +1,206 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+import pandas as pd
+
+
+class _Indexing(object):
+    def __init__(self, instance, axis=None):
+        self._obj = instance
+        self._axis = axis
+
+    def __call__(self, axis=None):
+        """Set the axis to index on."""
+        # verify axis param, discard value
+        self._obj._is_sequence_axis(axis)
+        return self.__class__(self._obj, axis=axis)
+
+    def __getitem__(self, indexable):
+        if self._axis is not None:
+            if self._obj._is_sequence_axis(self._axis):
+                return self._slice_on_first_axis(self._obj, indexable)
+            else:
+                return self._slice_on_second_axis(self._obj, indexable)
+
+        if type(indexable) is tuple:
+            if len(indexable) > 2:
+                raise ValueError("Can only slice on two axis. Tuple is length:"
+                                 " %r" % len(indexable))
+            elif len(indexable) > 1:
+                return self._handle_both_axes(*indexable)
+            else:
+                indexable, = indexable
+
+        return self._slice_on_first_axis(self._obj, indexable)
+
+    def _handle_both_axes(self, seq_index, pos_index):
+        seq_index = self._convert_ellipsis(seq_index)
+        pos_index = self._convert_ellipsis(pos_index)
+
+        if not hasattr(seq_index, '__iter__') and (seq_index is Ellipsis or
```

Comment:
Is the `Ellipsis` check necessary since you're calling `_convert_ellipsis(seq_index)`?